pkg/controller/v1beta1/inferenceservice/utils/utils.go

```diff
@@ -378,6 +378,11 @@ func UpdateImageTag(container *corev1.Container, runtimeVersion *string, serving
 				if !strings.HasSuffix(container.Image, "-gpu") {
 					container.Image = image + "-gpu"
 				}
+			} else if servingRuntime != nil && *servingRuntime == constants.VLLMServer {
+				// For vLLM server, use the GPU image from environment variable if GPU Enabled
+				if image, exists := utils.GetEnvVarValue(container.Env, "VLLM_GPU_IMAGE"); exists {
+					container.Image = image
```

Comment:
I am not sure if it is a good approach; ideally, we need to have VLLM CPU and GPU-specific runtimes, since the parameters and configurations aren't the same.